lib/twilio/preflight/preflight.ts

```diff
@@ -196,12 +202,43 @@ export class PreflightTest extends EventEmitter {
       fileInputStream: this._options.fakeMicInput ?
         this._getStreamFromFile() : undefined,
     });
+
+    // Device sets the loglevel so start logging after initializing the device.
+    // Then selectively log options that users can modify.
+    const userOptions = [
```

Comment:
It seems like we do this kind of thing in several places. Should we extract options behavior this into a separate module? Maybe in a future ticket?

Reply:
I think there's a lot of room for improving our options DX in general, I think the typing needs a bit of cleanup and perhaps the options overrides should be moved into their own module. Probably out of scope for this ticket. We can worry about it later I agree.